src/memory.rs

```diff
@@ -424,20 +384,33 @@ impl<'a, 'tcx> Memory<'a, 'tcx> {
         }
         return Ok(None);
     }
-}
 
-// The derived `Ord` impl sorts first by the first field, then, if the fields are the same
-// by the second field, and if those are the same, too, then by the third field.
-// This is exactly what we need for our purposes, since a range within an allocation
-// will give us all `Entry`s that have that `AllocId`, and whose `packed_start` is <= than
-// the one we're looking for, but not > the end of the range we're checking.
-// At the same time the `packed_end` is irrelevant for the sorting and range searching, but used for the check.
-// This kind of search breaks, if `packed_end < packed_start`, so don't do that!
-#[derive(Eq, PartialEq, Ord, PartialOrd)]
-struct Entry {
-    alloc_id: AllocId,
-    packed_start: u64,
-    packed_end: u64,
+    #[inline]
+    pub(crate) fn begin_unaligned_read(&mut self, aligned: bool) {
```

Comment:
I tried that. The trouble is, we often end up needing an `&mut EvalContext` inside the closure.

What I could try is to provide such an unaligned_read closure on `EvalContext`, rather than on `Memory`. It could still be implemented in `memory.rs` so as to have access to private fields.